sys/net/routing/rpl/rpl_control_messages.c

```diff
@@ -252,9 +248,18 @@ void rpl_init_root(rpl_options_t *rpl_opts)
 
 }
 
+/* TODO: this function is used by the ip layer for source routing (non-storing mode)
+ * to determine if the node is root and change the dodag direction (upwards -> downwards).
+ * This is a hack and needs to be refactored when the new network stack is ready.
+ */
 uint8_t rpl_is_root(void)
 {
-    return i_am_root;
+    for (uint8_t i = 0; i < RPL_MAX_DODAGS; i++) {
```

Comment:
to keep the code consistent it would be nice to iterate like you did for instance [here](https://github.com/RIOT-OS/RIOT/pull/2624/files#diff-2449d1ecd0a59722da77e45344a96c22R909)